### PR TITLE
[ARCTIC-558] Drop table operation will leave an empty table directory in hdfs

### LINKED ISSUE #558: [Bug]: Drop table operation will leave an empty table directory in hdfs
### What happened?

Drop table operation will leave an empty table directory on hdfs.
When you create a new table with the same name as the deleted table, the ams will throw an exception that the path already exists.

### Affects Versions

0.3.x

### What engines are you seeing the problem on?

Core

### How to reproduce

Drop an arctic table

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by shendanfengg**
Fix #558 

## Why are the changes needed?

* Now drop arctic table will leave an empty directory in hdfs

## Brief change log

  -After drop table, delete table directory 

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)

### CODE REVIEW

File: hive/src/main/java/com/netease/arctic/hive/catalog/ArcticHiveCatalog.java [hunk 1]
```diff
@@ -137,6 +136,7 @@ protected void doDropTable(TableMeta meta, boolean purge) {
     } catch (TException | InterruptedException e) {
       throw new RuntimeException("Failed to drop table:" + meta.getTableIdentifier(), e);
     }
+    super.doDropTable(meta, purge);
```
Comment (zhoujinsong) [inline hive/src/main/java/com/netease/arctic/hive/catalog/ArcticHiveCatalog.java:139]: Why moving droping hive table first?

**Review by zhoujinsong — COMMENTED** [review]
@shendanfengg  I left some comments.

File: hive/src/main/java/com/netease/arctic/hive/catalog/ArcticHiveCatalog.java (hunk 1 above)
Comment (shendanfengg) [reply hive/src/main/java/com/netease/arctic/hive/catalog/ArcticHiveCatalog.java:139]: When super.doDropTable is called and purge == true, all directories of the table will be remove. And then when try to drop hive table, the path will not be found

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**